Update pull_request_template.md (#2656)
dbt-labs/docs.getdbt.com: .github/pull_request_template.md

## What are you changing in this pull request and why?
<!---

> Describe your changes, and why you're making them. If linked to an open

Reviewer: ```suggestion
Describe your changes and why you're making them. If linked to an open
```

> issue or a pull request on dbt core, then link to them here! 

Reviewer: ```suggestion
issue or a pull request on dbt Core, then link to them here! 
```

To learn more about the writing conventions used in the dbt Labs docs, see the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md).
-->

## Checklist
<!--
Uncomment if you're publishing docs for a prerelease version of dbt (delete if not applicable):

> - [ ] Add versioning components, as described in [Versioning Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/versioningdocs.md)

Reviewer: Do we have a list of "must review" pages whenever a new version is released? If not, I can put something together.

Author: @matthewshaver I started a release checklist here: https://github.com/dbt-labs/docs.getdbt.com/pull/1355 I wonder if it would work better as its own doc in the Contributing folder?

- [ ] Add a note to the prerelease version [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/guides/migration/versions)
-->

> - [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [Versioning docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/versioningdocs.md) so my content adheres to these guidelines.

Reviewer: We appear to have deleted the versioningdocs.md document.

Author: D'oh! Good catch

> - [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [Versioning docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/versioningdocs.md) so my content adheres to these guidelines.

Author: ```suggestion
- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
```

- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [ ] Add page to `website/sidebars.js`

> - [ ] Provide a unique filename for the new page

Reviewer: Should we update this to include renaming pages and add a checklist item for redirects?

> Removing existing pages (delete if not applicable):

Reviewer: ```suggestion
Removing or renaming existing pages (delete if not applicable):
```
